src/iadmin.cpp

```diff
@@ -1620,8 +1620,22 @@ doCommand( char *cmdToken[], rodsArguments_t* _rodsArgs = 0 ) {
     }
 
     if ( strcmp( cmdToken[0], "mkgroup" ) == 0 ) {
-        generalAdmin( 0, "add", "user", cmdToken[1], "rodsgroup",
-                      "", "", "", "", "", "" );
+        const auto version = irods::to_version(Conn->svrVersion->relVersion);
+        const bool using_add_group = (version && version.value() > irods::version{4, 3, 3});
+        const char* entity_type = using_add_group ? "group" : "user";
+        // If we are adding a group, we don't need to specify a user type.
+        const char* user_type = using_add_group ? "" : "rodsgroup";
```

Comment:
Consider grouping these together via an if-else block to make it even clearer to readers that these are related.

Reply:
I think using an if-else block addressed all three comments. Please let me know if you had something else in mind.